### DISCUSSION TITLE
Automatically assign transcoding to specific clients

### DISCUSSION THREAD

**Discussion by Mikumikuer** [navidrome/navidrome#3464] [Ideas]
Some encoding(like tta) are not support by webrowser. 
Is it possible to detect user is logging with WebUI and auto assign transcoding method?

**Comment by metalheim** [comment]
Not possible at the moment, you can set transcoding for a specific client (user x and Browser). 
But navidrome will then transcode everything to that target format (also compatible files; even upsample lower-res files)

Whats currently possible is beeing mentioned here #2530 and here already #1592
Changing this behaviour needs changes to navidrome, that are tracked in these tickets:

https://github.com/navidrome/navidrome/issues/351#issuecomment-1357440141
#2051

**Reply by Mikumikuer** [reply]
Temporarily solved this issue by set trigger to database
At the 'player' table you can detect specific client and assign transcode by test 'client' field and assign 'transcoding_id' field, this will affect immediately to frontend.
As for the upsample problem, i add a flac transcode method and deleted bitrate parameters, this will force ffmpeg to use VBR, the transcoded file size is about same as original.

**Comment by prk0ghy** [comment]
This would be really helpful since some formats are not supported by default in the browser. I.e. .AIFF which can (in my setup) only be played when enabling transcoding for the WebUI player.
